common/lib/opentelemetry/common/http/request_attributes.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Common
+    module HTTP
+      # RequestAttributes contains instrumentation helpers for http instrumentation client requests
+      module RequestAttributes
+        QUERY_PARAM_START_KEY = '?'
+        def from_request(method:, config: {}, uri: nil, scheme: nil, target: nil, url: nil, hostname: nil, port: nil)
+          {
+            OpenTelemetry::SemanticConventions::Trace::HTTP_METHOD => method,
+            OpenTelemetry::SemanticConventions::Trace::HTTP_SCHEME => scheme || uri&.scheme,
+            OpenTelemetry::SemanticConventions::Trace::HTTP_TARGET => hide_query_params(target || uri&.request_uri, config),
+            OpenTelemetry::SemanticConventions::Trace::HTTP_URL => hide_query_params(url || uri&.to_s, config),
```

Comment:
Rather than serializing the `uri` and then searching for `?` and stripping the tail, if the `uri` is present, it might be more efficient to:
```ruby
uri = uri.dup
uri.query = nil
uri.to_s
```
or:
```ruby
uri = uri.dup
uri.query = nil
uri.request_uri
```
as appropriate.

Obviously, we also need to handle the case where `uri` is not present.